### PR TITLE
LibJS: Generate bytecode for throw statements

### PR DESCRIPTION
**Pull request by gunnarbeutner**
**LibJS: Generate bytecode for throw statements**

**LibJS: Stop bytecode execution after we've encountered an exception**

```
> console.log("Hello"); throw 3; console.log("World");
Hello
>
```

### CODE REVIEW

**Comment by awesomekling** [comment]
I suppose this is okay for a first step, but if there's a matching `catch` in the same `Executable`, we need to teleport there instead of abandoning execution.
I think the nicest way to achieve this would be to have `VM::throw_exception()` move the bytecode interpreter to the right PC (as opposed to checking for exceptions between each instruction)

File: Userland/Libraries/LibJS/Bytecode/Op.h [hunk 1]
```diff
@@ -395,6 +396,17 @@ class Decrement final : public Instruction {
     String to_string(Bytecode::Executable const&) const;
 };
 
+class Throw final : public Instruction {
+public:
+    Throw()
```
Comment (alimpfard) [inline Userland/Libraries/LibJS/Bytecode/Op.h:401]: Repeating what I said on discord, this should be a block terminator.

**Comment by Hendiadyoin1** [comment]
Maybe use a mechanism similar to the continueable scope, but a catching scope, that back propagates everything it does not catch and only the last one stops execution

**Comment by gunnarbeutner** [comment]
I've added a commit that deals with generating code for while/for/etc. loops after we've encountered `return` or `throw`:

**LibJS: Don't generate code after we've encountered a terminating op**

**Pull request merged**